kore/src/main/scala/org/kframework/kore/interface.scala

```diff
@@ -54,6 +53,7 @@ object K {
         case (_, _:KRewrite) => -1
         case (_:InjectedKLabel, _) => 1
         case (_, _:InjectedKLabel) => -1
+        case (_, _) => 0
```

Comment:
most of this looks good but this is definitely /not/ correct. Two arbitrary terms that don't match any of the above classes are /not/ equal. You should probably raise an error here.

Reply:
Now throwing a `KEMException.criticalError`